build/azure-pipelines/product-build.yml

```diff
@@ -171,10 +177,87 @@ stages:
             parameters:
               VSCODE_QUALITY: ${{ variables.VSCODE_QUALITY }}
 
+  - ${{ if eq(parameters.VSCODE_BUILD_TUNNEL_CLI, true) }}:
+    - stage: CompileCLI
+      dependsOn: []
+      jobs:
+        - ${{ if or(eq(parameters.VSCODE_BUILD_LINUX, true), eq(parameters.VSCODE_BUILD_LINUX_ALPINE, true)) }}:
+          - job: LinuxX86
+            pool: vscode-1es-linux
+            steps:
+              - template: ./linux/cli-build-linux.yml
+                parameters:
+                  VSCODE_QUALITY: ${{ variables.VSCODE_QUALITY }}
+                  VSCODE_CLI_DIR: $(Build.SourcesDirectory)/cli
+                  VSCODE_CLI_BINARY_NAME: ${{ variables.VSCODE_CLI_BINARY_NAME }}
+                  VSCODE_CLI_TARGETS:
+                    - ${{ if eq(parameters.VSCODE_BUILD_LINUX_ALPINE, true) }}:
+                      - { target: x86_64-unknown-linux-musl, artifact: vscode_cli_alpine_x64_cli }
+                    - ${{ if eq(parameters.VSCODE_BUILD_LINUX, true) }}:
+                      - { target: x86_64-unknown-linux-gnu, artifact: vscode_cli_linux_x64_cli }
+
```

Comment:
Instead of bringing CLI specific knowledge up to `product-build`, could we just forward all `VSCODE_BUILD_LINUX_*` parameters to `cli-build-linux.yml` and have that do the right thing in that pipeline?